src/Frontend/Assets.php

```diff
@@ -173,6 +179,10 @@ protected function makeLessCompiler(string $filename): LessCompiler
             $compiler->setFileSourceOverrides($this->fileSourceOverrides);
         }
 
+        $this->customFunctions = array_merge(resolve('flarum.frontend.default_preloads'), $this->customFunctions);
+
+        $compiler->setCustomFunctions($this->customFunctions);
```

Comment:
Why not `extend` the singleton and then inject the full set of custom functions into the compiler's constructor? I fear that this will end up overriding the functions added by other instances of the same extender.

Reply:
Mainly because I don't know how I'd go about doing that whatsoever.

Reply:
See https://github.com/flarum/core/blob/dc48e2327b1212c3b22fa57b23550f64b24632be/src/Extend/Auth.php#L63-L75  for an example of extending a singleton, and https://github.com/flarum/core/blob/dc48e2327b1212c3b22fa57b23550f64b24632be/src/Frontend/FrontendServiceProvider.php#L31-L31 for where `Assets` is instantiated (and so where to add constructor args).

Reply:
Think it should be done now...